### PR TITLE
[EventHubs] Doc warning on changing the value of the instance variable EventDataBatch.max_size_in_bytes

### PR DESCRIPTION
**Pull request by yunhaoling**
addressing issue: https://github.com/Azure/azure-sdk-for-python/issues/14752

concern: this is kina breaking change, should we do this?

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
lgtm

**Review by rakshith91 — APPROVED** [review]

File: sdk/eventhub/azure-eventhub/azure/eventhub/_common.py [hunk 1]
```diff
@@ -338,6 +338,10 @@ class EventDataBatch(object):
     **Please use the create_batch method of EventHubProducerClient
     to create an EventDataBatch object instead of instantiating an EventDataBatch object directly.**
 
+    **It is highly recommended not to update the value of the instance variable `max_size_in_bytes` on an
+    instantiated EventDataBatch object which might lead to sending failure as the Event Hub service
+    has a fixed quota on the maximum allowed size of events.**
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_common.py:343]: maybe something like this?
```suggestion
    **WARNING: Updating the value of the instance variable `max_size_in_bytes` on an instantiated EventDataBatch object is HIGHLY DISCOURAGED. The updated `max_size_in_bytes` value may conflict with the maximum size of events allowed by the Event Hubs service and result in a sending failure.**
```

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_common.py:343]: your wording is far more better than mine, I'll take yours :P
thanks!

**Review by swathipil — APPROVED** [review]
:shipit:

**Pull request merged**